### PR TITLE
added 0/1 knapsack in Java -> Dynamic-Programming. Issue #72

### PR DESCRIPTION
**Pull request by harshraj9988**
## 👨‍💻 Changes proposed

- Added code for 0/1 Knapsack (Dynamic Programming)
- Updated README.md
## ✔️ Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

Added 0-1 Knapsack (Dynamic-Programming) in Java

recursion -> memoization -> tabulation -> space optimzation

Starting with the naive recursive solution and gradually improving time complexity with memoization, then addressing the space complexity with tabulation and finally with space optimization.

Reorganized README.md and renamed 0/1 knapsack java file
Resolved merge conflict

(mentioned time complexity and space complexity of each approach)

### CODE REVIEW

**Comment by Kumar-laxmi** [comment]
@harshraj9988 Please resolve the **MERGE CONFLICT**

**Comment by harshraj9988** [comment]
Resolved Merge Conflict.
Kindly review again.

File: README.md [hunk 1]
```diff
@@ -35,9 +35,10 @@
  
 | No. | Algorithm Name                                                                        | Available languages                       | Category                                                                |
 | --- | ----------------------------------------------------------------------------------- | ------------------------------ | ------------------------------------------------------------------------- |
+
```
Comment (Kumar-laxmi) [inline README.md:38]: There some error in the table of README.md

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]

**Comment by Kumar-laxmi** [comment]
@harshraj9988 Please don't update the README.md as there has to be some changes in it, please remove updations of README.md from this PR

**Comment by harshraj9988** [comment]
sure! will remove README.md from this PR

**Pull request closed without merging**